src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DefaultDockerClient.java

```diff
@@ -32,6 +33,17 @@
 public class DefaultDockerClient {
     public static final Logger logger = LogManager.getLogger(DefaultDockerClient.class);
 
+    /**
+     * connect error messages.
+     */
+    private static final String READ_CONNECTION_TIME_OUT = "read: connection timed out";
+    private static final String NET_HTTP_TIMEOUT = "net/http";
```

Comment:
Can you revert this message to what it was before? We should avoid retrying indefinitely for errors that are not guaranteed to be fixed by retrying. Those errors should ideally have a finite retry count but for the purpose of this PR, just keep this a bit more specific

Reply:
> Can you revert this message to what it was before? We should avoid retrying indefinitely for errors that are not guaranteed to be fixed by retrying. Those errors should ideally have a finite retry count but for the purpose of this PR, just keep this a bit more specific

At the begining fo this ticket, I've asked for Micheal if I should try once or other retry rules, he told me "It should retry forever, as long as the thread is not interrupted".  So I did as this. 
Also you could give me a rule, which error should retry, retry  indefinitely or how many times.